tools/versioner/cmd/root.go

```diff
@@ -0,0 +1,219 @@
+// Copyright 2018 Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/tools/apidiff/delta"
+	"github.com/Azure/azure-sdk-for-go/tools/apidiff/exports"
+	"github.com/spf13/cobra"
+)
+
+var rootCmd = &cobra.Command{
+	Use:   "versioner <staging dir>",
+	Short: "Creates or updates the latest major version for a package from staged content.",
+	Long: `This tool will compare a staged package against its latest major version to detect
+breaking changes.  If there are no breaking changes the latest major version is updated
+with the staged content.  If there are breaking changes the staged content becomes the
+next latest major vesion and the go.mod file is updated.
+`,
+	Args: func(cmd *cobra.Command, args []string) error {
+		if err := cobra.ExactArgs(1)(cmd, args); err != nil {
+			return err
+		}
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cmd.SilenceUsage = true
+		return theCommand(args)
+	},
+}
+
+// Execute executes the specified command.
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
```

Comment:
do we want to print the error ?

Reply:
It will get printed as part of the `cobra` infrastructure.